### ISSUE TITLE
test_reporter.TestErrorReporting.test_timing uses clock and tests many things

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11014]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#1470|
|Type|defect|
|Created|2006-02-04 06:24:09Z|

<details><summary>Searchable metadata</summary>

```
trac-id__1470 1470
type__defect defect
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__tests tests
time__1139034249000000 1139034249000000
changetime__1143960061000000 1143960061000000
version__ 
owner__ 
cc__exarkun cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
test_timing uses the clock, which makes it unstable.
It also tests a few different things. Here are the author's (lifeless comments:

[11:15am] lifeless: well
[11:16am] lifeless: I think that it is testing that calls to
_somethingStarted|stopped return a time value, and that the time value is cached
[11:17am] lifeless: looking at the code it also tests that nested timings occur
[11:17am] lifeless: so its checking that when you get 'start, start, stop,
start, stop, stop', the last stop records the entire time
[11:18am] lifeless: or its sortof checking that but not all that clearly
[11:18am] lifeless: anyway, if i wanted to remote the clock dependency for it I
would allow the clock to be stubbed out - rather than time.time.time() in the
code, have self.time.time()
[11:19am] lifeless: and in the test case create a reporter and set reporter.time
to a test controlled clock
[11:19am] myouko left the chat room. (Read error: 104 (Connection reset by peer))
[11:19am] lifeless: I'd also trout lifeless for not making the test of the
nested aggregate time more clear and explicit.
[11:20am] lifeless: jml: ^^^
[11:21am] jml: lifeless: thanks
[11:21am] jml: lifeless: I'll blow it up into a few tests
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml commented|
|-|-|

```
#!html
<pre>
lifeless: I'd also probably factor out the timing code futher now that the using
code is cleaner
lifeless: and have a 'NestedTimer' class which is used by the reporter
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

Fix in [branches/timing-1470-2](branches/timing-1470-2).
Assigning to you because this also fixes 1538.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

Hooray, very nice.  Merge away.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [16528]) Cleanup Trial's timing support and tests thereof

Author: jml Reviewer: exarkun Fixes [#11014](https://github.com/twisted/twisted/issues/11014)

Factor out the bits that actually check the clock, then stub those bits out in the tests. This makes the tests more reliable -- particularly on Windows.

We also remove the nested timing abstraction, because we weren't ever nesting calls to timed things.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
